dpctl/tensor/_print.py

```diff
@@ -220,25 +223,41 @@ def print_options(*args, **kwargs):
         dpt.set_print_options(**options)
 
 
-def _nd_corners(x, edge_items, slices=()):
-    axes_reduced = len(slices)
-    if axes_reduced == x.ndim:
-        return x[slices]
-
-    if x.shape[axes_reduced] > 2 * edge_items:
-        return dpt.concat(
-            (
-                _nd_corners(
-                    x, edge_items, slices + (slice(None, edge_items, None),)
-                ),
-                _nd_corners(
-                    x, edge_items, slices + (slice(-edge_items, None, None),)
-                ),
-            ),
-            axis=axes_reduced,
+def _nd_corners(arr_in, edge_items):
+    _shape = arr_in.shape
+    res_shape = tuple(
+        2 * (edge_items + 1) if _shape[i] > 2 * (edge_items + 1) else _shape[i]
```

Comment:
```suggestion
        max_shape if _shape[i] > max_shape else _shape[i]
```

where `max_shape = 2 * edge_items + 1` defined before the `tuple` call.
and same `2*edge_items + 1` into a variable